src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -118,14 +163,66 @@ protected void processAccountingDocumentFromXml(String fileName, CreateAccountin
+            logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
+        } else if (reportItem.isNonBusinessRuleFailure()) {
             logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
         } else {
             logReport.getFilesSuccessfullyProcessed().add(fileName);
         }
         LOG.info("processAccountingDocumentFromXml: Value of reportItem.isNonBusinessRuleFailure just prior to method return =" + reportItem.isNonBusinessRuleFailure());
     }
-    
+
+    protected CreateAccountingDocumentFileEntry findExistingFileEntry(String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        Map<String, String> criteria = Map.of(KFSPropertyConstants.FILE_NAME, cleanedFileName);
+        Collection<CreateAccountingDocumentFileEntry> fileEntries = businessObjectService.findMatching(
+                CreateAccountingDocumentFileEntry.class, criteria);
+        return CollectionUtils.isNotEmpty(fileEntries) ? fileEntries.iterator().next() : null;
+    }
+
+    protected String getCleanedFileName(String fileName) {
+        String cleanedFileName = fileName;
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.BACKSLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.BACKSLASH);
+        }
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.SLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.SLASH);
+        }
+        cleanedFileName = StringUtils.lowerCase(cleanedFileName, Locale.US);
+        return cleanedFileName;
+    }
+
+    private boolean shouldExcludePreviouslyProcessedFiles() {
+        return parameterService.getParameterValueAsBoolean(KFSConstants.CoreModuleNamespaces.FINANCIAL, 
+                CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCOUNTING_DOCUMENT_SERVICE_COMPONENT_NAME, 
+                CuFPParameterConstants.CreateAccountingDocumentService.DUPLICATE_FILE_CHECK_IND);
+    }
+
+    private AccountingXmlDocumentListWrapper unmarshalAccountingDocumentFile(String fileName) {
+        byte[] fileData = LoadFileUtils.safelyLoadFileBytes(fileName);
+        return (AccountingXmlDocumentListWrapper) batchInputFileService.parse(
+                accountingDocumentBatchInputFileType, fileData);
+    }
+
+    private void createFileEntry(AccountingXmlDocumentListWrapper accountingXmlDocuments, String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        Long nextFileId = sequenceAccessorService.getNextAvailableSequenceNumber(
+                CreateAccountingDocumentConstants.FILE_ID_SEQUENCE_NAME);
+        CreateAccountingDocumentFileEntry fileEntry = new CreateAccountingDocumentFileEntry();
+        fileEntry.setFileId(nextFileId.toString());
+        fileEntry.setFileName(StringUtils.left(cleanedFileName, FileEntryFieldLengths.FILE_NAME));

```

Comment:
I think the file size clean should be included in getCleanedFileName() so the search will find entries of file names that had to be trimmed for length

Reply:
I'll refactor the cleaning and truncating of the filename accordingly.